### PR TITLE
Allow probing of point state by tag.

### PR DESCRIPTION
**Pull request by thorstenhater**
This concludes the introduction of 'labels' for synapse identification / placements started back in 
PR #1504 by @noraabiakar. Labels are now used when probing a synapse's state.

### CODE REVIEW

File: arbor/fvm_lowered_cell_impl.hpp [hunk 1]
```diff
@@ -913,17 +915,27 @@ void resolve_probe(const cable_probe_point_state& p, probe_resolution_data<B>& R
...
+            if (handle.mech_id != mech_id) return;
+            auto mech_index = handle.mech_index;
+            R.result.push_back(fvm_probe_scalar{{data + mech_index},
+                                                 point_info_of(target,
+                                                               lid,
+                                                               mech_index,
+                                                               synapses.at(mech),
+                                                               R.M.mechanisms.at(mech).multiplicity)});
+        }
+    }
```
Comment (boeschf) [inline arbor/fvm_lowered_cell_impl.hpp:938]: just for my understanding - can there ever be more than one probe resolved and pushed to result?

**Review by boeschf — COMMENTED** [review]
lgtm, thanks.

File: arbor/fvm_lowered_cell_impl.hpp (hunk 1 above)
Comment (thorstenhater) [reply arbor/fvm_lowered_cell_impl.hpp:938]: Formally, yes, as `synapse_ranges` is a multimap of ranges. This _could_ happen 
if two synapses are given the same label. We could forbid that, but currently don't.

Comment (boeschf) [reply arbor/fvm_lowered_cell_impl.hpp:938]: sure, sounds good

**Review by boeschf — APPROVED** [review]

File: arbor/fvm_lowered_cell_impl.hpp (hunk 1 above)
Comment (thorstenhater) [reply arbor/fvm_lowered_cell_impl.hpp:938]: Ah, that's impossible, as the probe contains the mechanism type (the id in the resolution step):
```
if (handle.mech_id != mech_id) return; 
```
So, multiple synapses of the **same** type can appear in the same probe, here.

**Pull request merged**